src/t8_geometry/t8_geometry_implementations/t8_geometry_occ.cxx

```diff
@@ -941,10 +1230,86 @@ t8_geometry_occ::t8_geom_edge_parameter_to_face_parameters(const int edge_index,
                                                                     first, last);
+  Handle_Geom_Surface surface = BRep_Tool::Surface(face);
   curve_on_surface->D0(edge_param, uv);
   face_params[0] = uv.X();
   face_params[1] = uv.Y();
+
+  /* Check for right conversion of edge to surface parameter and correct if needed */
+  /* Checking u parameter */
+  if (surface_params != NULL) {
+    double parametric_bounds[4];
+    surface->Bounds(parametric_bounds[0], parametric_bounds[1], parametric_bounds[2], parametric_bounds[3]);
+    if (surface->IsUClosed()) {
+      for (int i_face_node = 0; i_face_node < num_face_nodes; ++i_face_node) {
+        if (surface_params[i_face_node * 2] == parametric_bounds[0]) {
+          if (face_params[0] == parametric_bounds[1]) {
+            face_params[0] = parametric_bounds[0];
+          }
+        }
+        else if (surface_params[i_face_node * 2] == parametric_bounds[1]) {
+          if (face_params[0] == parametric_bounds[0]) {
+            face_params[0] = parametric_bounds[1];
+          }
+        }
+      }
+    }
+    /* Checking v parameter */
+    if (surface->IsVClosed()) {
+      for (int i_face_node = 0; i_face_node < num_face_nodes; ++i_face_node) {
+        if (surface_params[i_face_node * 2 + 1] == parametric_bounds[0]) {
+          if (face_params[1] == parametric_bounds[1]) {
+            face_params[1] = parametric_bounds[0];
+          }
+        }
+        else if (surface_params[i_face_node * 2 + 1] == parametric_bounds[1]) {
+          if (face_params[1] == parametric_bounds[0]) {
+            face_params[1] = parametric_bounds[1];
+          }
+        }
+      }
+    }
+  }
+}
+
+void
+t8_geometry_occ::t8_geom_get_surface_parametric_bounds(const int surface_index,
+                                                       double *bounds) const
+{
+  const Handle_Geom_Surface occ_surface = t8_geom_get_occ_surface(surface_index);
+  occ_surface->Bounds(bounds[0], bounds[1], bounds[2], bounds[3]);
+}
+
+void
+t8_geometry_occ::t8_geom_get_edge_parametric_bounds(const int edge_index,
+                                                    double *bounds) const
+{
+  const Handle_Geom_Curve occ_edge = t8_geom_get_occ_curve(edge_index);
+  bounds[0] = occ_edge->FirstParameter();
+  bounds[1] = occ_edge->LastParameter();
+}
+
+int
+t8_geometry_occ::t8_geom_check_if_edge_is_closed (int geometry_index) const
+{
+  const Handle_Geom_Curve occ_edge = t8_geom_get_occ_curve(geometry_index);
+  return occ_edge->IsClosed();
+}
+
+int
+t8_geometry_occ::t8_geom_check_if_surface_is_U_closed (int geometry_index) const
+{
+  const Handle_Geom_Surface occ_surface = t8_geom_get_occ_surface(geometry_index);
+  return occ_surface->IsUClosed();
+}
+
+int
+t8_geometry_occ::t8_geom_check_if_surface_is_V_closed (int geometry_index) const
+{
+  const Handle_Geom_Surface occ_surface = t8_geom_get_occ_surface(geometry_index);
+  return occ_surface->IsVClosed();
 }
```

Comment:
It would be better to add an argument and do u or v in the same function. This way the function becomes more useful in a loop, where you loop over u and v
`t8_geometry_occ::t8_geom_is_surface_closed (const int surface_index, const int dir) const`